src/main/java/org/folio/service/pieces/PiecesClaimingService.java

```diff
@@ -0,0 +1,237 @@
+        log.info("createJobsByVendor:: Returning claiming results, claimingResults: {}", JsonObject.mapFrom(claimingResults).encodePrettily());
+        return claimingResults;
+      });
+  }
+
+  private List<Future<List<String>>> createUpdatePiecesAndJobFutures(JsonObject config, Map<String, List<String>> pieceIdsByVendorId,
+                                                                     RequestContext requestContext) {
+    var updatePiecesAndJobFutures = new ArrayList<Future<List<String>>>();
+    pieceIdsByVendorId.forEach((vendorId, pieceIds) -> config.stream()
+      .filter(entry -> isExportTypeClaimsAndCorrectVendorId(vendorId, entry) && Objects.nonNull(entry.getValue()))
+      .forEach(entry -> {
+        log.info("createJobsByVendor:: Preparing job integration detail for vendor, vendor id: {}, pieces: {}, job key: {}",
+          vendorId, pieceIds.size(), entry.getKey());
+        updatePiecesAndJobFutures.add(updatePiecesAndCreateJob(pieceIds, entry, requestContext));
+      }));
+    return updatePiecesAndJobFutures;
+  }
+
+  private static ClaimingResults createEmptyClaimingResults(String message) {
+    return new ClaimingResults().withClaimingPieceResults(List.of(new ClaimingPieceResult().withError(new Error().withMessage(message))));
+  }
+
+  private List<ClaimingPieceResult> createSuccessClaimingResults(List<List<String>> updatedPieceLists) {
+    return updatedPieceLists.stream().flatMap(Collection::stream).distinct()
+        .map(pieceId -> new ClaimingPieceResult().withPieceId(pieceId).withStatus(SUCCESS))
+        .toList();
+  }
+
+  private List<ClaimingPieceResult> createErrorClaimingResults(Map<String, List<String>> pieceIdsByVendorId, String message) {
+    return pieceIdsByVendorId.values().stream()
+        .flatMap(Collection::stream)
+        .map(pieceId -> new ClaimingPieceResult().withPieceId(pieceId).withStatus(FAILURE).withError(new Error().withMessage(message)))
+        .toList();
+  }
+
+  private static boolean isExportTypeClaimsAndCorrectVendorId(String vendorId, Map.Entry<String, Object> entry) {
+    return entry.getKey().startsWith(String.format("%s_%s", EXPORT_TYPE_CLAIMS, vendorId));
+  }
+
+  private Future<List<String>> updatePiecesAndCreateJob(List<String> pieceIds, Map.Entry<String, Object> entry,
+                                                        RequestContext requestContext) {
+    log.info("updatePiecesAndCreateJob:: Updating pieces and creating a job, job key: {}, count: {}", entry.getKey(), pieceIds.size());
+    return pieceUpdateFlowManager.updatePiecesStatuses(pieceIds, PieceBatchStatusCollection.ReceivingStatus.CLAIM_SENT, requestContext)
+        .compose(v -> createJob(entry.getKey(), entry.getValue(), pieceIds, requestContext).map(pieceIds));
+  }
+
+  private Future<Void> createJob(String configKey, Object configValue, List<String> pieceIds, RequestContext requestContext) {
```

Comment:
Small suggestion: maybe lets pass params it self with better naming, it is hard to understand what is entry get Key and value is